socket/data.c

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file data.c
+ * @brief Data Representation Layer Implementation
+ * @date 2024-04-24
+ * @version 1.0
+ * @authors
+ * 	- TELLIER--CALOONE Tom
+ * 	- DELANNOY Anaël
+ */
+
+#include "data.h"
+
+/**
+ * @fn void send_stream_message(socket_t *exchange_socket, buffer_t content)
+ * @brief send a message on a stream socket
+ * @param exchange_socket: exchange socket to use for sending
+ * @param content: content to send
+ */
+void send_stream_message(socket_t *exchange_socket, buffer_t content) {
+	CHECK(write(exchange_socket->file_descriptor, content, strlen(content)), "Can't send STREAM message")
+}
+
+/**
+ * @fn void send_dgram_message(socket_t *exchange_socket, buffer_t content, char* ip, int port)
+ * @brief send a message on a datagram socket
+ * @param exchange_socket: exchange socket to use for sending
+ * @param content: content to send
+ * @param ip: sender's IP address
+ * @param port: sender's port
+ */
+void send_dgram_message(socket_t *exchange_socket, buffer_t content, char* ip, int port) {
```

Comment:
My version (made in class) is different. I don' know which is the best,  i will leave it up to you: 
"
void envoyerMessDGRAM(socket_t *sockEch, buffer_t buffer, char *ip, int port){

    //Création de l'adressage de destination 
    adr2struct(&sockEch->addrDst, ip, port);
    //Envoi du message à l'adresse de destination 
    CHECK(sendto(sockEch->fd, buffer, strlen(buffer)+1, 0, (struct sockaddr *)&sockEch->addrDst, sizeof(sockEch->addrDst)), "Can't send");
    //Fermeture du socket
    close(sockEch->fd);

}
"